### ISSUE TITLE
[FR] UI Full Customization

### ISSUE THREAD

**Issue by DeMoN100** [AppFlowy-IO/AppFlowy#2847]
### Description

Have premade themes as choice selection is tacky & dull to follow the typical motto. As an open source alternative people like to customize their own UI with workspaces so lets make it happen. Rewrite appearance to be a CSS selector so a person can select their own colors for the UI from a color picker or input by color code.

### Impact

Makes people happy to have a nortion alternative where they can customize their own UI. I like mine black and red maybe you like yours terminal green theme or perhaps you like purple and yellow together.

### Additional Context

Make it happen!

**Comment by annieappflowy** [comment]
cc: @a-wallen

**Comment by a-wallen** [comment]
Hi @DeMoN100 thanks for contributing to this issue! I'm working on adding customizable themes:
https://github.com/AppFlowy-IO/AppFlowy/pull/2670
This is the MVP. Please message me on discord at a-wallen to provide suggestions. I look forward to working with you.

**Comment by DeMoN100** [comment]
Looks pretty interesting but I don't see any GUI for selecting colors. If theres an export import then thats cool but the themes should be easily creatable in a gui. It would also be nice if in appearance there was a text box so each individual can put in their own css for extra styling.

As for making a theme the GUI should have everything in a list with display of what it affects. You should be able to click on it then get a color picker like the example below.
![f310d4e3-cdc2-45bb-a647-3ca7019268c0](https://github.com/AppFlowy-IO/AppFlowy/assets/8564572/9d67c021-fd75-492e-b33c-13714b361672)

With the options to save & export/import a person can style a theme import it. For easy sharing there could be a text box with the code ready to copy on click so you can just paste it to anyone. The import/export options should spit out .txt files with the theme's code including extra css options.

The main goal is to create a gui so non-coders can happily stylize their appflowy the way they like. The extra css text box that would appear in appearance settings would allow someone to tweak their style even further. For example with that someone could round icons, add gradients anywhere, customize every page a different color, add outer glow to text & more.

People will love it & it will be another + under the apps belt. I stick with apps that let you customize & more foss apps are going that direction to give the user one more thing they can control. I hope this will be accomplished in appflowy.

@a-wallen Thanks but I don't use discord I use Matrix for iming.

**Comment by a-wallen** [comment]
@DeMoN100 I think that those all sound like great ideas. We could really use all the feedback we can get before we land this. Could you keep in touch on this thread? I'll keep you updated when I make changes based on your feedback. Thank you so much, we value your opinion!

**Comment by DeMoN100** [comment]
> @DeMoN100 I think that those all sound like great ideas. We could really use all the feedback we can get before we land this. Could you keep in touch on this thread? I'll keep you updated when I make changes based on your feedback. Thank you so much, we value your opinion!

Yes I will if you decide to check out matrix one day it would be cool if you could make a bridge to your discord support so users across matrix can get help too. You can find my room in my social links. https://chaotic.gg/@truedemon100 It's always the best to extend your reach!

I look forward to seeing more features & implementations into this app. Theres so much that needs to be done to get it up to par with notion but the foundation of the groundwork is by far going to blow them away.

**Comment by DeMoN100** [comment]
I hope the idea is still being built into the app :D looking forward to tweaking appflowy to my own style.
